sidecar/internal/service/reclaimspace.go

```diff
@@ -61,12 +97,18 @@ func (rs *ReclaimSpaceServer) ControllerReclaimSpace(
 	pvName := req.GetPvName()
 	klog.Info(pvName)
 
-	// TODO: get following params using kubeClient and pvName
+	pv, err := rs.kubeClient.CoreV1().PersistentVolumes().Get(context.TODO(), pvName, metav1.GetOptions{})
+	volID := pv.Spec.CSI.VolumeHandle
+	volattributes := pv.Spec.CSI.VolumeAttributes
+
+	deviceSecret, err := getSecret(rs.kubeClient,
+		pv.Spec.CSI.NodeStageSecretRef.Namespace,
```

Comment:
you will need to check if `pv.Spec.CSI.NodeStageSecretRef != nil`. Not all CSI-drivers set this. Only create the `deviceSecret` if it is non-empty.